src/client/datascience/kernel-launcher/kernelLauncher.ts

```diff
@@ -140,4 +184,32 @@ export class KernelLauncher implements IKernelLauncher {
             kernel_name: kernelConnectionMetadata.kernelSpec?.name || 'python'
         };
     }
+
+    // If we need to install our dependencies now
+    // then install ipykernel into the interpreter or throw error
+    private async installDependenciesIntoInterpreter(interpreter: PythonEnvironment, cancelToken?: CancellationToken) {
+        // Cache the install question so when two kernels start at the same time for the same interpreter we don't ask twice
+        let deferred = this.dependencyPromises.get(interpreter.path);
+        if (!deferred) {
+            deferred = createDeferredFromPromise(
+                this.kernelDependencyService.installMissingDependencies(interpreter, cancelToken)
+            );
+            this.dependencyPromises.set(interpreter.path, deferred);
+        }
+
+        // Get the result of the question
+        try {
+            const result = await deferred.promise;
+            if (result !== KernelInterpreterDependencyResponse.ok) {
+                throw new Error(
```

Comment:
Can we have a custom error, this way we can deal with this in a different way (higher up) if required.
E.g. `IPyKernelNotInstalledError`
Also feel having custom errors is better than just `Error`, feels too generic.

Only suggesting so that we can have special code tomorrow if required, and feels like a good practice

Reply:
Not a mandatory change.

Reply:
On second thought I thinmk we'll need it, so that we can ignore this error higher up.

Reply:
No it's not necessary for ignoring